core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -151,12 +151,14 @@ public ZoneFailureDetector apply(final String input) {
      */
     protected final Object mutex = new Object[0];
 
-    private static final Function<Collection<Entity>, Entity> defaultRemovalStrategy = new Function<Collection<Entity>, Entity>() {
-        @Override public Entity apply(Collection<Entity> contenders) {
```

Comment:
Agreed, let's be paranoid. Mark it as deprecated, with a comment saying why we have not deleted it.

There are two ways changing your anonymous inner class can break persistence: one is if someone has persisted it; two is if deleted or moving the class changes the classnames of other anonymous inner classes (there are 5 of them in DynamicClusterImpl)!